src/Text/Parsing/Parser/String.purs

```diff
@@ -121,6 +125,14 @@ oneOf ss = satisfy (flip elem ss) <?> ("one of " <> show ss)
 noneOf :: forall m. Monad m => Array Char -> ParserT String m Char
 noneOf ss = satisfy (flip notElem ss) <?> ("none of " <> show ss)
 
+-- | Match one of the Unicode characters in the array.
+oneOfCodePoints :: forall m. Monad m => Array CodePoint -> ParserT String m CodePoint
+oneOfCodePoints ss = satisfyCodePoint (flip elem ss) <|> defer \_ -> fail $ "Expected one of " <> show (singleton <$> ss)
```

Comment:
I chose to make error messages deferred to avoid the cost of generating the error messages until they're needed. Feels like a small thing, but the waste just bothers me too much.

Let me know if this is undesirable for some reason, I'll change it back to `<?>`

Reply:
Thanks, I think this is a great idea, and from the tests you added it clearly works. We should probably do this everywhere. I opened https://github.com/purescript-contrib/purescript-parsing/issues/128